### PR TITLE
Onboard NVIDIA Driver Container to PublishContainer script

### PR DESCRIPTION
**Pull request by henryli001**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- This PR adds implementation for publishing the NVIDIA GPU driver container

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=607560&view=results

### CODE REVIEW

**Review by mandeepsplaha — APPROVED** [review]

**Review by mandeepsplaha — APPROVED** [review]

File: .pipelines/containerSourceData/scripts/PublishContainers.sh [hunk 1]
```diff
@@ -336,6 +338,21 @@ do
...
                 "$ARCHITECTURE_TO_BUILD"
+        elif "$IS_NVIDIA_GOLDEN_IMAGE"; then
+            # For nvidia images, we need to create multi-arch tag for the following format:
+            # E.g. for azurelinuxpreview.azurecr.io/nvidia/cuda/driver:550-5.15.160.1-1.cm2-mariner2.0.2.0.20240626-amd64
+            # the multi-arch tag to create is: azurelinuxpreview.azurecr.io/nvidia/cuda/driver:550-5.15.160.1-1.cm2-mariner2.0
+            multiarch_tag=${container_tag%.*.*.*}
+
+            azure_linux_version=$(awk -F '-' '{print $4}' <<< "$container_tag")              # 550-5.15.160.1-1.cm2-mariner2.0.2.0.20240626 -> mariner2.0.2.0.20240626
+            azure_linux_version=$(awk -F '.' '{print $3"."$4}' <<< "$azure_linux_version")   # [mariner2].[0].[2].[0].[20240626] -> 2.0
+            #                                                                                                  ^   ^
```
Comment (dallasd1) [inline .pipelines/containerSourceData/scripts/PublishContainers.sh:349]: Was there a comment here

File: .pipelines/containerSourceData/scripts/PublishContainers.sh [hunk 2]
```diff
@@ -336,6 +338,21 @@ do
                 "$major_version" \
                 "$azure_linux_version" \
                 "$ARCHITECTURE_TO_BUILD"
+        elif "$IS_NVIDIA_GOLDEN_IMAGE"; then
+            # For nvidia images, we need to create multi-arch tag for the following format:
+            # E.g. for azurelinuxpreview.azurecr.io/nvidia/cuda/driver:550-5.15.160.1-1.cm2-mariner2.0.2.0.20240626-amd64
+            # the multi-arch tag to create is: azurelinuxpreview.azurecr.io/nvidia/cuda/driver:550-5.15.160.1-1.cm2-mariner2.0
+            multiarch_tag=${container_tag%.*.*.*}
+
+            azure_linux_version=$(awk -F '-' '{print $4}' <<< "$container_tag")              # 550-5.15.160.1-1.cm2-mariner2.0.2.0.20240626 -> mariner2.0.2.0.20240626
```
Comment (dallasd1) [inline .pipelines/containerSourceData/scripts/PublishContainers.sh:347]: [nit] this is a bit confusing to follow because of the same name being used

**Review by dallasd1 — APPROVED** [review]

File: .pipelines/containerSourceData/scripts/PublishContainers.sh (hunk 1 above)
Comment (henryli001) [reply .pipelines/containerSourceData/scripts/PublishContainers.sh:349]: Yes, this line is a comment that includes the two "^" on the right.

File: .pipelines/containerSourceData/scripts/PublishContainers.sh (hunk 2 above)
Comment (henryli001) [reply .pipelines/containerSourceData/scripts/PublishContainers.sh:347]: I was following the same pattern of obtaining the Azure Linux version as what was being done for HCI and golden container images down below in the same file. If this is not considered a major problem, I can file a bug on myself to take care of this naming issue here later in a separate change.

**Review by dallasd1 — APPROVED** [review]

**Pull request merged**